src/DevLicenseMeta.sol

```diff
@@ -9,47 +10,75 @@ import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
 /**
  * @title DevLicenseMeta
  * @dev Extends DevLicenseStake to add metadata functionality, including image and description handling.
+ * @dev To facilitate potential upgrades, this agreement employs the Namespaced Storage Layout (https://eips.ethereum.org/EIPS/eip-7201)
  */
-contract DevLicenseMeta is DevLicenseStake {
-    string public _imageToken;
-    string public _imageContract;
-    string public _descriptionToken;
-    string public _descriptionContract;
+contract DevLicenseMeta is Initializable, DevLicenseStake {
+    /// @custom:storage-location erc7201:DIMOdevLicense.storage.DevLicenseMeta
+    struct DevLicenseMetaStorage {
+        string _imageToken;
+        string _imageContract;
+        string _descriptionToken;
+        string _descriptionContract;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("DIMOdevLicense.storage.DevLicenseMeta")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant DEV_LICENSE_META_STORAGE_LOCATION =
+        0x528ba85f43aa21e96244176d4db284a510484bc2715e2bea8ee3cbf67d5fb800;
+
+    function _getDevLicenseMetaStorage() internal pure returns (DevLicenseMetaStorage storage $) {
+        assembly {
+            $.slot := DEV_LICENSE_META_STORAGE_LOCATION
+        }
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
 
     /**
      * @dev Sets initial metadata for tokens and the contract itself.
-     * @param receiver_ Address to receive payments.
-     * @param licenseAccountFactory_ Factory address for license account creation.
-     * @param provider_ Address of the price provider.
-     * @param dimoTokenAddress_ Address of the DIMO token.
-     * @param dimoCreditAddress_ Address of the DIMO credit.
-     * @param licenseCostInUsd_ Cost of a license in USD.
+     * @param image String of a SVG for contract and token URIS
+     * @param description String with a description for contract and token URIs
+     */
+    function __DevLicenseMeta_init(string calldata image, string calldata description) internal onlyInitializing {
+        __DevLicenseStake_init();
+
+        DevLicenseMetaStorage storage $ = _getDevLicenseMetaStorage();
+
+        $._imageToken = Base64.encode(bytes(image));
+        $._imageContract = Base64.encode(bytes(image));
+
+        $._descriptionToken = description;
+        $._descriptionContract = description;
+    }
+
+    /**
+     * @notice Returns the image token
```

Comment:
The function name and description read a bit oddly to me. "Token image"? Are these required for some standard?

Reply:
Yeah that is odd. Not required as far as I know. It is just to have the image in a variable.